tests/Unit/Amqp/AMQPTimestampTest.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/*
+ * PHP AMQP-Compat - php-amqp/ext-amqp compatibility.
+ * Copyright (c) Dan Phillimore (asmblah)
+ * https://github.com/asmblah/php-amqp-compat/
+ *
+ * Released under the MIT license.
+ * https://github.com/asmblah/php-amqp-compat/raw/main/MIT-LICENSE.txt
+ */
+
+declare(strict_types=1);
+
+namespace Asmblah\PhpAmqpCompat\Tests\Unit\Amqp;
+
+use AMQPTimestamp;
+use AMQPValueException;
+use ArgumentCountError;
+use Asmblah\PhpAmqpCompat\Tests\AbstractTestCase;
+use TypeError;
+
+/**
+ * Class AMQPTimestampTest.
+ *
+ * @author Dan Phillimore <dan@ovms.co>
+ */
+class AMQPTimestampTest extends AbstractTestCase
+{
```

Comment:
I'm not asking you to change this. Just thinking out loud: Here you could have used 2 generators and write only 2 tests. Although the way you wrote it it's more readable

Reply:
Cheers - I did consider it, but I thought it kept the test code simpler this way as there are only a few scenarios and not much setup code 👍